File: app/src/main/java/com/shentu/wallpaper/mvp/contract/SplashContract.java
package com.shentu.wallpaper.mvp.contract;

import com.jess.arms.mvp.IModel;
import com.jess.arms.mvp.IView;
import com.shentu.wallpaper.model.entity.SplashAd;
import com.shentu.wallpaper.model.response.SplashAdResponse;

import io.reactivex.Observable;

public interface SplashContract {
    interface View extends IView {
        void showSplash(SplashAd splashAd);

        void toMainPage();

        void startCountDown(int total);

        void showCountDownText(int time);
    }

    interface Model extends IModel {
        Observable<SplashAdResponse> getSplashAd();
    }
}

File: app/src/main/java/com/shentu/wallpaper/mvp/contract/SplashContract.kt
package com.shentu.wallpaper.mvp.contract

import com.jess.arms.mvp.IModel
import com.jess.arms.mvp.IView
import com.shentu.wallpaper.model.entity.MicroUser
import com.shentu.wallpaper.model.entity.SplashAd
import com.shentu.wallpaper.model.response.BaseResponse
import com.shentu.wallpaper.model.response.SplashAdResponse
import io.reactivex.Observable

interface SplashContract {
    interface View : IView {
        fun toMainPage()
        fun startCountDown(total: Int)
        fun showCountDownText(time: Int)
    }

    interface Model : IModel {

    }
}